cartoframes/core/cartodataframe.py

```diff
@@ -115,76 +118,88 @@ def to_carto(self, *args, **kwargs):
 
-            .. code::
+        if not inplace:
+            return frame
 
-                from cartoframes import CartoDataFrame
+    def set_geometry_from_xy(self, x, y, drop=False, inplace=False, crs=None):
+        if isinstance(x, str) and x in self and isinstance(y, str) and y in self:
+            x_col = self[x]
+            y_col = self[y]
+        else:
+            x_col = x
+            y_col = y
 
-                cdf = CartoDataFrame.from_file('filename.csv')
-                cdf.convert(geom_column='my_geom_column')
+        # Generate geometry
+        geom_col = points_from_xy(x_col, y_col)
 
-            Passing lnglat_columns explicitly:
+        # Call super set_geometry with generated column
+        frame = super(CartoDataFrame, self).set_geometry(geom_col, inplace=inplace, crs=crs)
 
-            .. code::
+        if drop:
+            if frame is None:
+                frame = self
+            del frame[x]
+            del frame[y]
 
-                from cartoframes import CartoDataFrame
+        return frame
 
-                cdf = CartoDataFrame.from_file('filename.csv')
-                cdf.convert(lnglat_columns=['longitude', 'latitude'])
+    @property
+    def _constructor(self):
+        return CartoDataFrame
 
-            Passing the index column explicitly:
+    def __getitem__(self, key):
+        result = super(self._constructor, self).__getitem__(key)
+        if isinstance(key, (list, ndarray, Series)):
+            result.__class__ = self._constructor
+        return result
 
-            .. code::
+    def __finalize__(self, *args, **kwargs):
+        result = super(self._constructor, self).__finalize__(*args, **kwargs)
+        result.__class__ = self._constructor
+        return result
 
-                from cartoframes import CartoDataFrame
+    def astype(self, *args, **kwargs):
+        result = super(self._constructor, self).astype(*args, **kwargs)
+        result.__class__ = self._constructor
+        return result
 
-                cdf = CartoDataFrame.from_file('filename.csv')
-                cdf.convert(index_column='my_index')
+    def merge(self, *args, **kwargs):
+        result = super(self._constructor, self).merge(*args, **kwargs)
+        result.__class__ = self._constructor
+        return result
 
-        """
-        generate_index(self, index_column, drop_index)
-        generate_geometry(self, geom_column, lnglat_columns, drop_geom, drop_lnglat)
-        return self
+    def dissolve(self, *args, **kwargs):
+        result = super(self._constructor, self).dissolve(*args, **kwargs)
+        result.__class__ = self._constructor
+        return result
 
-    def viz(self, *args, **kwargs):
-        """
-        Creates a :py:class:`Map <cartoframes.viz.Map>` visualization
-        """
-        from ..viz import Map, Layer
-        return Map(Layer(self, *args, **kwargs))
+    def explode(self, *args, **kwargs):
```

Comment:
General comment: you could add the parent (geopandas') docstring back to preserve the information in there (eg something like `explode.__doc__ = geopandas.GeoDataFrame.explode.__doc__`, or a decorator that can automate this)

Reply:
Cool. We will take this into account for the next release. Thanks!